lib/puppet/provider/package/yum.rb

```diff
@@ -175,7 +175,15 @@ def install
         operation = :install
       end
       should = nil
-    when true, false, Symbol
+    when true, :present, :installed 
+      # if we have been given a source and we were not asked for a specific
+      # version feed it to yum directly
+      if @resource[:source]
+        wanted = @resource[:source]
+        self.debug "Downloading directly from #{wanted}"
```

Comment:
Since this could be a local file (as opposed to an HTTP source) I'd like to update it to read "Installing directly from #{wanted}". We'll also want to mark the string for externalization.